packages/mds-ingest-service/repository/index.ts

```diff
@@ -291,6 +270,71 @@ class IngestReadWriteRepository extends ReadWriteRepository {
     }
   }
 
+  private buildCursor = (cursor: VehicleEventsQueryParams) =>
+    Buffer.from(JSON.stringify(cursor), 'utf-8').toString('base64')
+
+  private parseCursor = (cursor: string): VehicleEventsQueryParams & { limit: number } => {
+    try {
+      return JSON.parse(Buffer.from(cursor, 'base64').toString('utf-8'))
+    } catch (error) {
+      throw new ValidationError('Invalid cursor', error)
+    }
+  }
```

Comment:
It's risky to state that just because it decodes & json parses that it is a `VehicleEventsQueryParams & { limit: number }`. I *heavily* recommend passing this through an AJV validator before associating the type with it, especially cause this is right at the DB level.

Reply:
Would still like to see this, but I'm not gonna block the PR on it. If you decide to merge without it, would you mind adding a ticket to the backlog so we can add validation at some point?